.editorconfig

```diff
@@ -0,0 +1,12 @@
+; EditorConfig file: https://EditorConfig.org
+; Install the "EditorConfig" plugin into your editor to use
+
+root = true
+
+[*]
+charset = utf-8
+end_of_line = lf
+insert_final_newline = true
+indent_style = space
+indent_size = 2
+trim_trailing_whitespace = true
```

Comment:
I use prettier defaults in this project.

Reply:
Unfortunately, I don't know of any functional IDE tools which leverage existing config from projects like prettier (or ESLint which has a prettier plugin) so as to do pre-linting--i.e., to set the indent when hitting return (rather than linting on save, along with other fixes). [EditorConfig](https://editorconfig.org/), on the other hand, is well supported (and works with non-JS also).

Reply:
>  I don't know of any functional IDE tools

Any IDE support them. Via plugins or builtin. Even VIM. Which IDE do you use?

Reply:
Atom

Reply:
And I am referring to project-aware setting of the indent.

Reply:
https://atom.io/packages/prettier-atom

Many projects in js community use prettier. Adding editorconfig is adding another tool for formatting by your personal preference.

Reply:
Thank you, and I am aware of that Atom plugin (I have it installed). But note how it says, "format on save". It makes the changes either manually or when you save. It doesn't know instantly when you hit enter that it should immediately indent to the proper number of spaces or tabs. The EditorConfig plugin does do this.

Anyways, not a big deal, and your call of course. But EditorConfig currently fills a different purpose than Prettier, at least given the absence of tools which set the indent immediately upon hitting enter, and not waiting until save. There are plenty of projects which use both EditorConfig and ESLint, for example, even though both having settings for indent spaces. Ideally, ESLint tools would do so, but the only solution I've seen ( https://github.com/stuart-williams/eslint-tab-length ) does not work too well for me (it toggles the system default).